library/common/main_interface.cc

```diff
@@ -88,7 +88,15 @@ envoy_status_t reset_stream(envoy_stream_t stream) {
 }
 
 envoy_status_t set_preferred_network(envoy_network_t network) {
-  Envoy::Network::MobileUtility::setPreferredNetwork(network);
+  envoy_network_t previous = Envoy::Network::Configurator::setPreferredNetwork(network);
+  if (previous != network && previous != ENVOY_NET_GENERIC) {
+    if (auto e = engine()) {
+      e->dispatcher().post([network]() -> void {
+        if (auto e = engine())
```

Comment:
I wonder if in the future we want to have Engine::networkChange(). If we want to have more things happen when a network changes. I am sure you are already thinking about that.

Reply:
We could also fire an event on network change in that function. Which I think we should do anyway to have a picture of when the engine was "aware" of a network change.

Reply:
Yes, I have thought about that. The current form felt reasonable enough to me for now, but we might be getting close to something like that making sense.

Reply:
Up to you. I think at least we should fire an (network_change with the new network) event.

Reply:
Up to you. I think at least we should fire an (network_change with the new network) event.